### DISCUSSION TITLE
How do I restart xrdp enough so that new Linux group memberships are reflected?

### DISCUSSION THREAD

**Discussion by bo-qeye** [neutrinolabs/xrdp#3043] [Q&A]
If I add a user to a group, that group membership does not show up, even if close down the terminal program completely.

I tried a clumsy restart of the VNC session for the particular user, but to no avail.

This is Rocky Linux 8 with stock GNOME.

The new group membership does show up fine if I ssh to the XRDP host.

**Comment by matt335672** [comment]
For xrdp v0.9.x you'll probably need to restart xrdp. This is because the user is authenticated in the main xrdp process with PAM (bug #1684). The PAM session is then not closed in that process, but in a sub-process.

This is fixed in xrdp v0.10 (currently in beta).

**Reply by bo-qeye** [reply]
Thanks for your reply, @matt335672. This xrdp restart would affect all users, right?

**Reply by matt335672** [reply]
Yes it would.

You'll ned to schedule the restart with your users.
